### PR TITLE
Cask: refuse to trash root-owned files

### PR DESCRIPTION
**Pull request by amyspark**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [X] Have you successfully run `brew style` with your changes locally?
- [X] Have you successfully run `brew tests` with your changes locally?

-----

These files cannot be trashed from CLI without sudo, which trashes them
instead to root's Trash. AppleScript isn't of use here because it will
break Travis (as it will fail when asking for admin permissions).

Fixes homebrew/homebrew-cask#69897

### CODE REVIEW

**Comment by reitermarkus** [comment]
Maybe output the failed ones to `stderr` and then use that to display a list. Or subtract the ones in `stdout` from the original list.

File: Library/Homebrew/cask/artifact/abstract_uninstall.rb [hunk 1]
```diff
@@ -362,7 +362,13 @@ def uninstall_trash(*paths, **options)
       def trash_paths(*paths, command: nil, **_)
         return if paths.empty?
 
-        result = command.run!("/usr/bin/swift", args: [TRASH_SCRIPT, *paths])
+        trashable, untrashable = paths.partition(&:writable?)
+        unless untrashable.empty?
+          opoo "These files cannot be moved to the user's Trash: "
+          puts untrashable
```
Comment (reitermarkus) [inline Library/Homebrew/cask/artifact/abstract_uninstall.rb]: `$stderr.puts`

File: Library/Homebrew/cask/artifact/abstract_uninstall.rb [hunk 2]
```diff
@@ -362,7 +362,13 @@ def uninstall_trash(*paths, **options)
       def trash_paths(*paths, command: nil, **_)
         return if paths.empty?
 
-        result = command.run!("/usr/bin/swift", args: [TRASH_SCRIPT, *paths])
+        trashable, untrashable = paths.partition(&:writable?)
+        unless untrashable.empty?
+          opoo "These files cannot be moved to the user's Trash: "
```
Comment (reitermarkus) [inline Library/Homebrew/cask/artifact/abstract_uninstall.rb]: Trailing space here.

**Comment by amyspark** [comment]
Ping for review @reitermarkus .

**Comment by reitermarkus** [comment]
Thanks, @amyspark!

**Pull request merged**